google/cloud/aiplatform/training_jobs.py

```diff
@@ -3346,3 +3352,265 @@ def _run(
         )
 
         return model
+
+
+class AutoMLTextTrainingJob(_TrainingJob):
+    def __init__(
+        self,
+        display_name: str,
+        prediction_type: str,
+        multi_label: Optional[bool] = None,
+        sentiment_max: Optional[str] = None,
+        model_hint: str = "default",
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Constructs a AutoML Text Training Job.
+
+        Args:
+            display_name (str):
+                Required. The user-defined name of this TrainingPipeline.
+            prediction_type (str):
+                The type of prediction the Model is to produce, one of:
+            TODO
+            multi_label (bool):
+                Required and only applicable for text classification task. If false, a single-label (multi-class) Model will be trained (i.e.
+                assuming that for each text snippet just up to one annotation may be
+                applicable). If true, a multi-label Model will be trained (i.e.
+                assuming that for each text snippet multiple annotations may be
+                applicable).
+            sentiment_max (int):
+                Required and only applicable for sentiment task. A sentiment is expressed as an integer
+                ordinal, where higher value means a more
+                positive sentiment. The range of sentiments that
+                will be used is between 0 and sentimentMax
+                (inclusive on both ends), and all the values in
+                the range must be represented in the dataset
+                before a model can be created.
+                Only the Annotations with this sentimentMax will
+                be used for training. sentimentMax value must be
+                between 1 and 10 (inclusive).
+            project (str):
+                Optional. Project to run training in. Overrides project set in aiplatform.init.
+            location (str):
+                Optional. Location to run training in. Overrides location set in aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to run call training service. Overrides
+                credentials set in aiplatform.init.
+        """
+        super().__init__(
+            display_name=display_name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+        self._prediction_type = prediction_type
+        self._multi_label = multi_label
+        self._sentiment_max = sentiment_max
+        self._model_hint = model_hint
+
+        training_task_definition: str
+        training_task_inputs_dict: proto.Message
+
+        if self._prediction_type == "classification":
```

Comment:
Since there are no switch statements, is there a better way than to do this in python?

I don't quite like this as it couples each prediction_type with each other into one if-else statement.
May refactor to decouple them.

Reply:
Well, since we've flattened the hierarchy, the coupling is hard to avoid.

Reply:
May I suggest something like

```
        mapping = {
            "classification": {
                "task": schema.training_job.definition.automl_text_classification_task,
                "func": AutoMlTextClassificationInputs,
                "required_field": "multi_label"
                },
           "extraction": {
                "task": schema.training_job.definition.automl_text_extraction_task,
                "func": AutoMlTextExtractionInputs,
                "required_field": ""
                },
            "sentiment": {
                "task": schema.training_job.definition.automl_text_sentiment_task,
                "func": AutoMlTextSentimentInputs,
                "required_field": "sentiment_max"
            }
        }

        if self._prediction_type not in mapping:
            raise ValueError(
                "Prediction type must be one of 'classification', 'extraction', or 'sentiment'."
            )

        prediction_type = mapping[self._prediction_type]
        training_task_definition = prediction_type["task"]
        training_func = prediction_type["func"]
        required_field = prediction_type["required_field"]
        args = {}

        if required_field:
            self_required_field = self.__getattribute__("_" + required_field)

            # If there is a field required and it is missing
            if self_required_field is None:               
                raise ValueError("Required parameter missing:" + prediction_type["required_field"]
                    )
            args = {required_field: self_required_field}

        training_task_inputs_dict = training_func(**args).to_value()
```

Reply:
Thanks for contributing! There are some limitations to this particular solution though, such as only supporting a single `self_required_field` and requiring certain variables being named a certain way ('the underscore prefix').

I think in this case, the if-else makes things most explicit whereas preserving flexibility.

Reply:
The 'correct' approach would just be to unflatten the function signature and just encapsulate all required parameters into its own class (i.e. use the protos). However, for UX reasons it's been decided that flattening the various prediction types (and respective parameters) was preferable.

Reply:
To your first comment -- it would be pretty simple to support a list of fields: 
`"required_fields": ["multi_label"]"`. 

If there are no fields, it would simply be an empty list.  

If we cannot use protos in the solution, I think having some kind of mapping is the next best option.  I think the value of using an approach like this is that it is more readable - both in terms of logic and whitespace.  It can be hard to follow so many if/else statements, especially if they have nested if statements.   It also results in having a large amount of code indented far to the right, which means you have to start breaking up strings and functions, etc, to keep it under 80 characters.

Reply:
Thanks, those are great points. I'll discuss it with the team as this issue appears across a variety of our training jobs and might be useful in other places.